docu/Concepts/TechnicalRequirements/UC_Introduction_of_Gradido-ID.md

```diff
@@ -0,0 +1,141 @@
+| type            | int    | Defines the type of contact entry as enum: Email, Phone, etc                                                                                                           |
+| userID          | int    | Defines the foreign key to the `Users` table                                                                                                                         |
+| email           | String | defines the address of a contact entry of type Email                                                                                                                   |
+| phone           | String | defines the address of a contact entry of type Phone                                                                                                                   |
+| contactChannels | String | define the contact channel as comma separated list for which this entry is confirmed by the user e.g. main contact (default), infomail, contracting, advertisings, ... |
+
+### Database-Migration
+
+After the adaption of the database schema and to keep valid consistent data, there must be several steps of data migration to initialize the new and changed columns and tables.
+
+#### Initialize GradidoID
+
+In a one-time migration create for each entry of the `Users `tabel an unique UUID (version4).
+
+#### Primary Email Contact
+
+In a one-time migration read for each entry of the `Users `table the `Users.id` and `Users.email` and create for it a new entry in the `UsersContact `table, by initializing the contact-values with:
+
+* id = new technical key
+* type = Enum-Email
+* userID = `Users.id`
+* email = `Users.email`
+* phone = null
+* usedChannel = Enum-"main contact"
+
+and update the `Users `entry with `Users.emailId = UsersContact.Id` and `Users.passphraseEncryptionType = 1`
+
+After this one-time migration the column `Users.email` can be deleted.
+
+### Adaption of BusinessLogic
+
+The following logic or business processes has to be adapted for introducing the Gradido-ID
+
+#### Read-Write Access of Users-Table especially Email
+
+The ORM mapping has to be adapted to the changed and new database schema.
+
+#### Registration Process
+
+The logic of the registration process has to be adapted by
+
+* initializing the `Users.userID` with a unique UUID
+* creating a new `UsersContact `entry with the given email address and *maincontact* as `usedChannel `
+* set `emailID `in the `Users `table as foreign key to the new `UsersContact `entry
+* set `Users.passphraseEncrpytionType = 2` and encrypt the passphrase with the `Users.userID` instead of the `UsersContact.email`
+
+#### Login Process
+
+The logic of the login process has to be adapted by
+
+* search the users data by reading the `Users `and the `UsersContact` table with the email (or alias as soon as the user can maintain his profil with an alias)   as input
+* depending on the `Users.passphraseEncryptionType` decrypt the stored password
```

Comment:
This process should also update the password, since the password reaches us in plaintext (super insecure in my opinion, but apparently web-standard).

For security we could use something like SRP6. See [Wikipedia](https://en.wikipedia.org/wiki/Secure_Remote_Password_protocol) [Stanford](http://srp.stanford.edu/design.html)
This would be quite secure and quite applicable to a banking software, but holds quite some complexity.
We then would no longer transmit the password in plain text over the network, but an proof, that we own that password.

Reply:
that's right and should be done in the future - see #1811